### PR TITLE
Add file types for fabric.mod.json and mixins.json

### PR DESCRIPTION
**Pull request by Draylar**
Added file types for the main Fabric mod JSON file and mixin config JSONs. The only new functionality is special icons, but it can be expanded on later:

image

### CODE REVIEW

**Comment by natanfudge** [comment]
I would like to get Cloak approved and working before I pull this, so it might take a few days, sorry.

File: src/main/kotlin/cloak/idea/fileTypes/FabricModConfigFileType.kt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package cloak.idea.fileTypes
+
+import cloak.idea.util.CommonIcons
+import com.intellij.json.JsonFileType
+import com.intellij.json.JsonLanguage
+
+object FabricModConfigFileType : JsonFileType(JsonLanguage.INSTANCE) {
+    override fun getName() = "Fabric Mod Config"
```
Comment (natanfudge) [inline src/main/kotlin/cloak/idea/fileTypes/FabricModConfigFileType.kt:8]: Nice!

File: src/main/kotlin/cloak/idea/util/Icons.kt [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
 import com.intellij.openapi.ui.Messages
+import com.intellij.openapi.util.IconLoader
 
 object CommonIcons {
     val Error = Messages.getErrorIcon()
     val Warning = Messages.getWarningIcon()
     val Info = Messages.getInformationIcon()
     val Question = Messages.getQuestionIcon()
+
+    val Fabric = IconLoader.getIcon("/assets/fabric_logo_16.png");
```
Comment (natanfudge) [inline src/main/kotlin/cloak/idea/util/Icons.kt]: Create another object `FabricIcons` (in the same file) and put these there.

File: src/main/resources/assets/mixin_config.svg [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+        <polygon fill="#f5e942" fill-opacity=".8" points="7 1 3 5 7 5"/>
+        <path fill="#f5e942" fill-opacity=".8" d="M10.5957,12.9404 C10.5957,13.0624 10.5847,13.9504 10.5907,13.9754 C10.5907,13.9744 10.5927,13.9734 10.5927,13.9734 C10.6227,13.9734 11.4997,14.2654 11.4997,14.2654 C11.4997,14.2654 12.3717,13.9764 12.4247,13.9404 C12.4177,13.9274 12.4047,13.0624 12.4047,12.9404 C12.4047,12.4914 12.4937,11.9654 12.8087,11.5034 C12.5397,11.1134 12.4047,10.6304 12.4047,10.0594 C12.4047,9.9374 12.4147,9.0494 12.4087,9.0244 C12.4087,9.0254 12.4077,9.0254 12.4077,9.0254 C12.3767,9.0254 11.4997,8.7344 11.4997,8.7344 C11.4997,8.7344 10.6277,9.0234 10.5747,9.0584 C10.5817,9.0724 10.5957,9.9374 10.5957,10.0594 C10.5957,10.5084 10.5057,11.0334 10.1917,11.4964 C10.4597,11.8854 10.5957,12.3684 10.5957,12.9404"/>
```
Comment (natanfudge) [inline src/main/resources/assets/mixin_config.svg:5]: Hmmm yesss

**Review by natanfudge — APPROVED** [review]

File: src/main/resources/assets/mixin_config.svg (hunk 3 above)
Comment (Draylar) [reply src/main/resources/assets/mixin_config.svg:5]: I can fully assure you this was written by hand without any help

**Pull request merged**